data_steward/vocabulary.py

```diff
@@ -1,11 +1,179 @@
+OMOP_VOCABULARY_CONCEPT_ID = '44819096'
+
+ERROR_APPENDING = 'Appending to {in_path} which already contains rows for ' + AOU_GEN
+
+
+def format_date_str(s):
+    """
+    Format a date string to yyyymmdd if it is not already
+    :param s: the date string
+    :return: the formatted date string
+    """
+    if re.match(BQ_DATE_REGEX, s):
+        return s
+    elif re.match(RAW_DATE_REGEX, s):
+        parts = s[0:4], s[4:6], s[6:8]
+        return '-'.join(parts)
+    else:
+        raise ValueError('Cannot parse value {v} as date'.format(v=s))
+
+
+def _transform_csv(in_fp, out_fp, err_fp=None):
+    if not err_fp:
+        err_fp = sys.stderr
+    csv_reader = csv.reader(in_fp, delimiter=DELIMITER)
+    header = next(csv_reader)
+    date_indexes = []
+    for i in range(0, len(header)):
+        if header[i].endswith('_date'):
+            date_indexes.append(i)
+    csv_writer = csv.writer(out_fp, delimiter=DELIMITER, lineterminator=LINE_TERMINATOR)
+    csv_writer.writerow(header)
+    for row_index, row in enumerate(csv_reader):
+        try:
+            for i in date_indexes:
+                row[i] = format_date_str(row[i])
+            csv_writer.writerow(row)
+        except Exception, e:
+            message = 'Error %s transforming row:\n%s' % (e.message, row)
+            err_fp.write(message)
+
+
+def transform_file(file_path, out_dir):
+    """
+    Format file date fields and standardize line endings a local csv file and save result in specified directory
+
+    :param file_path: Path to the csv file
+    :param out_dir: Directory to save the transformed file
+    """
+    file_name = os.path.basename(file_path)
+    table_name, _ = os.path.splitext(file_name.lower())
+    out_file_name = os.path.join(out_dir, file_name)
+    err_dir = os.path.join(out_dir, ERRORS)
+    err_file_name = os.path.join(err_dir, file_name)
+    if not os.path.exists(err_dir):
+        os.makedirs(err_dir)
+    with open(file_path, 'rb') as in_fp, open(out_file_name, 'wb') as out_fp, open(err_file_name, 'wb') as err_fp:
+        _transform_csv(in_fp, out_fp, err_fp)
+
+
+def transform_files(in_dir, out_dir):
+    """
+    Transform vocabulary files in a directory and save result in another directory
+
+    :param in_dir: Directory containing vocabulary csv files
+    :param out_dir: Directory to save the transformed file
+    """
+    fs = os.listdir(in_dir)
+    for f in fs:
+        in_path = os.path.join(in_dir, f)
+        transform_file(in_path, out_dir)
+
+
+def get_aou_general_version():
+    return hash_dir(AOU_GENERAL_PATH)
+
+
+def get_aou_general_vocabulary_row():
+    aou_gen_version = get_aou_general_version()
+    # vocabulary_id vocabulary_name vocabulary_reference vocabulary_version vocabulary_concept_id
+    return DELIMITER.join([AOU_GEN, AOU_GEN, AOU_GEN_VOCABULARY_REFERENCE, aou_gen_version, AOU_GEN_VOCABULARY_CONCEPT_ID])
```

Comment:
It's fine for vocabulary_id and vocabulary_name to be the same, but this assumes they are always the same and uses the same variable to set them.  Would it be better to go ahead and separate them with a `AOU_GEN_ID` and `AOU_GEN_NAME` variables?